### DISCUSSION TITLE
`Cannot call a class as a function` after upgrading to next 13.5.6

### DISCUSSION THREAD

**Discussion by JacobWeisenburger** [vercel/next.js#57428] [Help] [answered]
### Summary

I upgraded to next 13.5.6 today and dev mode works fine, but when I build my app, I get a bunch of `Cannot call a class as a function` errors in the browser console (see below). I tried clicking on all the links in the error message but none of them seem to be helpful, because it's all minified. I have no idea how to debug this. Please help.

<details>
  <summary>See screenshots</summary>

![image](https://github.com/vercel/next.js/assets/31667350/c181b712-265b-4192-9cb8-c2e569356086)

![image](https://github.com/vercel/next.js/assets/31667350/c36da602-bc44-4f25-b67e-5a7d157cfb41)

![image](https://github.com/vercel/next.js/assets/31667350/c5eb86a1-ace2-421b-a45a-ba4e69cbd6c5)
</details>

**Comment by icyJoseph** [comment]
Do you see the same error if you turn off minification?

Open next.config.js and add `swcMinify:false`

**Reply by JacobWeisenburger** [reply]
that didn't change anything. It's still being minified.

#### next.config.mjs
```ts
// @ts-check

/**
 * Run `build` or `dev` with `SKIP_ENV_VALIDATION` to skip env validation.
 * This is especially useful for Docker builds.
 */
!process.env.SKIP_ENV_VALIDATION && ( await import( './src/env.mjs' ) )

/** @type {import('next').NextConfig} */
export default {
    swcMinify: false,
    webpack ( config ) {
        return {
            ...config,
            cache: false
        }
    },
    images: {
        remotePatterns: [
            { hostname: 'cdn05.carsforsale.com' },
            { hostname: 'lh3.googleusercontent.com' },
            { hostname: 'upload.wikimedia.org' },
        ],
    },
    async redirects () {
        return [
            { source: '/', destination: '/car-list', permanent: false },
            { source: '/cars', destination: '/car-list', permanent: false },
            { source: '/carlist', destination: '/car-list', permanent: false },
            { source: '/car%20list', destination: '/car-list', permanent: false },
        ]
    },
}
```

**Reply by icyJoseph** [reply]
Mmm I was expecting this to be an SWC issue. You still see the minification, but now it is minified with another tool... but that still triggers this error right?

I think, to prevent minification, to some extend, you need to mutate or add this to the webpack config:

`config.optimization.minify = false`

**Reply by JacobWeisenburger** [reply]
```
Build error occurred
ValidationError: Invalid configuration object. Webpack has been initialized using a configuration object that does not match the API schema.
 - configuration.optimization has an unknown property 'minify'.
```

```ts
/** @type {import('next').NextConfig} */
export default {
    swcMinify: false,
    webpack ( config ) {
        config.optimization.minify = false
        return {
            ...config,
            cache: false
        }
    },
}
```

**Reply by JacobWeisenburger** [reply]
I think you meant `config.optimization.minimize = false`. I tried this also, but the output is still minified

**Reply by icyJoseph** [reply]
Then it is probably React production code that's at fault here... But how to prove and fix this 🤔

**Reply by JacobWeisenburger** [reply]
One of the things I could make out from the minified code:

the code in the screenshot below seems to be directly correlated to [app-index.tsx](https://github.com/vercel/next.js/blob/462b8585b61b83fa079f40e2d8432cc643bf0713/packages/next/src/client/app-index.tsx#L16C1-L30C3)

![image](https://github.com/vercel/next.js/assets/31667350/b5d30c2e-b854-4d5c-904a-65e758679bf2)

Not sure if this is helpful or not, but just thought I would put it out there.

**Reply by icyJoseph** [reply]
This is failing locally right? With a production build anyway.

Could you remove routes until you get a green build? Trying to see if we can create a minimal reproduction case

**Reply by JacobWeisenburger** [reply]
I was hoping not to have to do that, but I guess that seems to be the only way. thanks for your help.

**Reply by icyJoseph** [reply]
Yeah well, I suppose you don't use class components right? Those are not supported as server components, but without looking at code I couldn't tell much more than that ~ i haven't had time to upgrade to the latest next though.

**Reply by JacobWeisenburger** [reply, marked as answer]
thanks for all your help. after spending way too much time looking into it, :D, I figured out that the problem had something to do with lodash, so I switched it to radash and everything seems to be working now.

![image](https://github.com/vercel/next.js/assets/31667350/be5121d5-ff86-42be-abea-2b20c2a82f1c)

**Reply by icyJoseph** [reply]
Oh wow, haha 😂 wonder what's going on inside Lodash then... I guess that's just another reason to avoid it altogether

**Reply by emilmilow** [reply]
@icyJoseph what made you think that `swcMinify:false` could solve the problem? Because I had the exact same error in my production builds ("Cannot call a class as a function") and was able to solve it with `swcMinify:false`.

**Reply by icyJoseph** [reply]
That's a good question, I guess mostly just experience. I have seen a couple of issues where, things work on dev mode, but somehow in production builds, things start to misbehave, because suddenly something is missing.

When the breakage is like that, a good check can be to disable SWC and see if that fixes it. If so then I try to make an SWC playground example, and then it gets fixed upstream.

These are usually, a code transformation for sake of smaller, more efficient code, that doesn't transform correctly. They tend to be rare edge cases though.

**Reply by icyJoseph** [reply]
We have found the reason.

- https://github.com/vercel/next.js/discussions/58182#discussioncomment-7514351

One has to opt out of the new automatic server side minification.
